### PR TITLE
[Core] Support relative path in Polling Strategy 

### PR DESCRIPTION
**Pull request by mssfang**
Based on the request of https://github.com/Azure/azure-sdk-for-java/issues/29375

Added the support for relative path in Polling Strategy.

### CODE REVIEW

**Review by weidongxu-microsoft — COMMENTED** [review]
Overall LGTM.

Wait for review from core experts.

File: sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java [hunk 1]
```diff
@@ -100,10 +119,12 @@ public Mono<PollResponse<T>> onInitialResponse(Response<?> response, PollingCont
...
+            pollingContext.setData(operationLocationHeaderName,
+                getAbsolutePath(operationLocationHeader.getValue(), endpoint, LOGGER));
         }
         if (locationHeader != null) {
-            pollingContext.setData(PollingConstants.LOCATION, locationHeader.getValue());
+            pollingContext.setData(PollingConstants.LOCATION,
+                getAbsolutePath(locationHeader.getValue(), endpoint, LOGGER));
         }
         pollingContext.setData(PollingConstants.HTTP_METHOD, response.getRequest().getHttpMethod().name());
         pollingContext.setData(PollingConstants.REQUEST_URL, response.getRequest().getUrl().toString());
```
Comment (haolingdong-msft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java:130]: Do we need to override getResult() to make get final result url absolute path as well?

Comment (weidongxu-microsoft) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java:130]: `response.getRequest().getUrl()` is from a working request and must be absolute.

File: sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java [hunk 2]
```diff
@@ -258,6 +259,61 @@ public void operationLocationPollingStrategyFailsOnPoll() {
         assertEquals(1, activationCallCount[0]);
     }
 
+    @Test
+    public void operationLocationPollingStrategyRelativePath() {
+        int[] activationCallCount = new int[1];
+        activationCallCount[0] = 0;
+        String endpointUrl = "http://localhost";
+        String mockPollRelativePath = "/poll";
+        String finalResultAbsolutePath = endpointUrl + "/final";
```
Comment (weidongxu-microsoft) [inline sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java:268]: the case in Haoling's comment is from deviceupdate that the final resource URL is also relative

e.g.
```
"resourceLocation": "/deviceUpdate/sdkinstance/updates/providers/fabrikam/names/vacuum/versions/2022.609.510.21?api-version=2022-07-01-preview"
```

Comment (weidongxu-microsoft) [reply sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java:268]: @mssfang Any update, as code freeze is closing?

Comment (mssfang) [reply sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java:268]: Sorry for the late reply. Do you mean resourceLocation could be either an absolute or relative path?

Comment (weidongxu-microsoft) [reply sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java:268]: Yes, in the case of deviceupdate backend, we found it is also relative (the ref on earlier comment is the real one we get).

File: sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java (hunk 1 above)
Comment (mssfang) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java:130]: updated

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM.

@alzimmermsft @srnagar for review from core.

**Review by alzimmermsft — APPROVED** [review]

**Review by srnagar — APPROVED** [review]

**Pull request merged**